### ISSUE TITLE
Strange cuda illegal memory allocation error

### ISSUE THREAD

**Issue by Davidyao99** [pytorch/pytorch#84445]
### 🐛 Describe the bug

I am trying to run the following GitHub repository.

https://github.com/neuroailab/EISEN

 However, I am receiving a strange Cuda illegal memory allocation error. 
![image](https://user-images.githubusercontent.com/52120948/188027404-12098a5c-bcc2-4af9-ba7b-ceaaa01ab117.png)
It does not seem to be the code since the repo author states that the code is running fine on his machine. It will be great if someone can help to verify if this is specific to my machine or not. Any idea if this is an issue with my Cuda/torch version? I have tried to debug this but was not successful. I did find that the error seems to come from this line: `indices = sample_inds[-1].view(B, T * K).unsqueeze(-1).expand(-1, -1, D)` from core/utils/utils.py line 97.

When trying to print the max value of the indices variable, it also throws the cuda illegal memory allocation error. The error is gone when I remove the `.expand(-1, -1, D)`, but it leads to a dimension issue further down the code.

The following is my environment.
![image](https://user-images.githubusercontent.com/52120948/188026199-9b1d8661-0709-44df-bb0e-0b505158d586.png)
I am also running this on wsl2 with the following nvidia-smi output:
![image](https://user-images.githubusercontent.com/52120948/188027473-74c14413-b74b-473d-b599-72d4052aa1c8.png)

### Versions

![image](https://user-images.githubusercontent.com/52120948/188027978-e81a3420-bcee-48c2-8396-48a8fd850a8a.png)

**Comment by eqy** [comment]
That seems somewhat unexpected, if you could check the input shapes and values of the indices and provide a repro script we can check if it's reproducible with with an updated PyTorch version.

**Comment by Davidyao99** [comment]
I can print the value of indices, but when I call tensor.max() on the indices, it leads to the cuda error...

As for reproduction, I simply followed the steps on https://github.com/neuroailab/EISEN, setting up my conda environment using the environment.yml and downloading the flow, dataset and models. I did not change any code.

**Comment by eqy** [comment]
Right, could you check if running `tensor.max()` on the same inputs in a standalone script reproduces this issue? That would help in debugging.
